go/roothash/api/commitment/pool.go

```diff
@@ -251,15 +252,23 @@ func (p *Pool) addOpenExecutorCommitment(
 		return ErrBadExecutorCommitment
 	}
 
-	if err := sv.VerifyTxnSchedulerSignature(body.TxnSchedSig, blk.Header.Round); err != nil {
+	if err := sv.VerifyTxnSchedulerSigner(body.TxnSchedSig, blk.Header.Round); err != nil {
 		logger.Debug("executor commitment has bad transaction scheduler signer",
 			"node_id", id,
 			"round", blk.Header.Round,
 			"err", err,
 		)
 		return err
 	}
-	if ok := body.VerifyTxnSchedSignature(blk.Header); !ok {
+	sigCtx, err := ProposedBatchSignatureContext.WithSuffix(p.Runtime.ID.String())
+	if err != nil {
+		logger.Error("preparing proposed batch signature context error",
+			"runtime_id", p.Runtime.ID,
+			"err", err,
+		)
+		return fmt.Errorf("proposed batch signature context error: %w", err)
+	}
+	if ok := body.VerifyTxnSchedSignature(sigCtx, blk.Header); !ok {
```

Comment:
Maybe it would be better if `VerifyTxnSchedSignature` took a runtime ID and generated the context internally?

Reply:
yeah agreed